aten/src/ATen/native/Convolution.cpp

```diff
@@ -539,6 +539,10 @@ struct ConvParams {
   }
   bool use_nnpack(const at::Tensor& input, const at::Tensor& weight) const  {
 #if AT_NNPACK_ENABLED()
+    if (!at::globalContext().userEnabledNNPACK()) {
+      return false;
+    }
+
     return at::_nnpack_available() &&
```

Comment:
Nit (short circuit ensures that only first rule will be avaluated)
```suggestion
    return at::globalContext().userEnabledNNPACK() && at::_nnpack_available() &&
```